### ISSUE TITLE
AssemblyReference.GetReferencedItems can be improved

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#6245]
Make FindSolutionItem not try to do IO on the path every time.

```
0.06%   GetReferencedItem  •  950 ms  •  1,000 calls  •  MonoDevelop.Projects.AssemblyReference.GetReferencedItem(Solution)
  0.06%   FindSolutionItem  •  948 ms  •  1,000 calls  •  MonoDevelop.Projects.Solution.FindSolutionItem(String)
    0.06%   FindSolutionItem  •  947 ms  •  16,100 calls  •  MonoDevelop.Projects.SolutionFolder.FindSolutionItem(String)
       0.04%   GetFullPath  •  634 ms  •  62,692 calls  •  System.IO.Path.GetFullPath(String)
       0.01%   9 functions hidden  •  165 ms total  •  389545 calls total
       0.01%   MoveNext  •  109 ms  •  75,310 calls  •  System.Collections.Immutable.ImmutableList+Enumerator`1.MoveNext
     0.00%   get_RootFolder  •  0 ms  •  1,000 calls  •  MonoDevelop.Projects.Solution.get_RootFolder
   0.00%   2 functions hidden  •  2 ms total  •  2000 calls total
```

> VS bug [#700045](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/700045)

**Comment by Therzok** [comment]
Already been fixed in master, seems like vs-sync failed to close this: https://github.com/mono/monodevelop/pull/6247

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-10] Comment by **xamarinc**



Fixed in version 7.7.0.1567 (release-7.7)



Pull Request #6259 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 6e360f9652120391296e56108e106a50368e1d5a (mono/monodevelop)
